src/utils/daos_hdlr.c

```diff
@@ -2554,6 +2554,11 @@ fs_copy(struct file_dfs *src_file_dfs,
 	if (dst_file_dfs->obj != NULL || dst_file_dfs->fd != -1)
 		file_close(dst_file_dfs, filename);
 out:
+	if (rc != 0) {
```

Comment:
I don't think this is the right fix.  I think we need to make sure we don't overwrite a failure rc with a success rc on line 2566.  So this gets rid of the memory leak but doesn't actually fix the bug that caused it.

Reply:
I see. How about this pattern?
```
int close_rc;

close_rc = file_closedir(src_file_dfs, src_dir);
if (close_rc != 0) {
        fprintf(stderr, "Could not close '%s': %d\n",
                   dir_name, close_rc);
        if (rc == 0)
                rc = close_rc;
}
```
FWIW, I have a refactor PR that simplifies this altogether by refactoring individual parts of this function into separate functions.

Reply:
I pushed a commit for this so you can see it in code